plugins/data-provider-email/classes/DataProvider/Email.php

```diff
@@ -149,32 +151,32 @@ public function fetch($limit = FALSE)
 						{
 							if ($part->subtype == 'HTML')
 							{
-								$html_message .= imap_fetchbody($connection, $email_number, $part_number);
+								$html_message .= imap_fetchbody($connection, $email->uid, $part_number, FT_UID);
 							} elseif ($part->subtype == 'PLAIN') {
-								$message .= imap_fetchbody($connection, $email_number, $part_number);
+								$message .= imap_fetchbody($connection, $email->uid, $part_number, FT_UID);
 							}
 						}
 					}
 					else
 					{
 						// or just fetch the body if not a multipart message
-						$message = imap_body($connection, $email_number);
+						$message = imap_body($connection, $email->uid, FT_UID);
 					}
 
 
 					// Process the email
 					if (! empty($html_message)) {
 						$html_message = imap_qprint($html_message);
-						$this->_process_incoming($overview[0], $html_message);
+						$this->_process_incoming($email, $html_message);
 					}
 					elseif (! empty($message))
 					{
 						$message = imap_qprint($message);
-						$this->_process_incoming($overview[0], $message);
+						$this->_process_incoming($email, $message);
 					}
 
 					// After processing, delete!
-					imap_delete($connection, $email_number);
+					//imap_delete($connection, $email_number);
```

Comment:
So this is dependent on imap setup, for gmail it doesn't work since they block delete you have to do a convoluted move operation to trash. For the others it just doesn't seem to work.

No we're not changing anything about the messages, we can mark them read but I kind of wanted to leave it as much as possible.